Initial transparent long stack trace support

CaptEmulation opened the pull request:
unfortunately, some necessary unit tests which should be passing are failing to satisfy `2.2.7.2 f either onFulfilled or onRejected throws an exception e, promise2 must be rejected with e as the reason.` from https://promisesaplus.com/

neonphog commented:
created https://github.com/CaptEmulation/ppp/pull/1

neonphog commented on lib/ppp.js:
```diff
@@ -160,6 +167,19 @@
+        //
+        // At the same time, cut off the first line; it's always just
+        // "[object Promise]\n", as per the `toString`.
+        this.stack = e.stack.substring(e.stack.indexOf("\n") + 1);
```
We may need to not do this... our version of opera doesn't include the error in the first line, so we might be deleting a line of stacktrace...

CaptEmulation replied:
I'll check

CaptEmulation commented:
- Unit tests pass
 - Removed the stripping of the first line of exception due to Opera
 - Added internal stack stripping support for Chrome

Does not seem to work for minified code.  I don't see that as a problem.

neonphog commented:
> Does not seem to work for minified code. I don't see that as a problem.

I'm not sure why, but after I tweaked a couple unrelated things, the minified stack traces started working. Maybe you forgot to `grunt uglify` after doing a fix?

Thanks for this, I think this is going to be super helpful!!

CaptEmulation commented:
> Maybe you forgot to grunt uglify after doing a fix?

probably
